From 4013bcc328d1a901637d662f04a60ef0ccc16d44 Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Sun, 20 Sep 2020 21:01:07 +0530
Subject: [PATCH] Enable dumpstate service 1.1

This patch will enable the dumpstate service 1.1
DevelopOptions depends on this service if we are enabling
this HAL.

Change-Id: I527f51c48846c271a622dafcecf1c051fab8fa2e
Tracked-On: OAM-92709
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 dumpstate/1.1/default/Android.bp                      |  8 ++++----
 .../default/android.hardware.dumpstate@1.1-service.rc |  8 ++++++++
 .../android.hardware.dumpstate@1.1-service.xml        | 11 +++++++++++
 3 files changed, 23 insertions(+), 4 deletions(-)
 create mode 100644 dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.rc
 create mode 100644 dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.xml

diff --git a/dumpstate/1.1/default/Android.bp b/dumpstate/1.1/default/Android.bp
index bd920752b..701d1c6f8 100644
--- a/dumpstate/1.1/default/Android.bp
+++ b/dumpstate/1.1/default/Android.bp
@@ -1,13 +1,13 @@
 cc_binary {
-    name: "android.hardware.dumpstate@1.1-service.example",
+    name: "android.hardware.dumpstate@1.1-service",
     vendor: true,
     relative_install_path: "hw",
     defaults: ["hidl_defaults"],
     init_rc: [
-        "android.hardware.dumpstate@1.1-service.example.rc",
+        "android.hardware.dumpstate@1.1-service.rc",
     ],
     vintf_fragments: [
-        "android.hardware.dumpstate@1.1-service.example.xml",
+        "android.hardware.dumpstate@1.1-service.xml",
     ],
     srcs: ["main.cpp"],
     shared_libs: [
@@ -21,6 +21,6 @@ cc_binary {
         "libutils",
     ],
     cflags: [
-        "-DLOG_TAG=\"android.hardware.dumpstate@1.1-service.example\"",
+        "-DLOG_TAG=\"android.hardware.dumpstate@1.1-service\"",
     ],
 }
diff --git a/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.rc b/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.rc
new file mode 100644
index 000000000..7eef64a79
--- /dev/null
+++ b/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.rc
@@ -0,0 +1,8 @@
+service dumpstate-1-1 /vendor/bin/hw/android.hardware.dumpstate@1.1-service
+    class hal
+    user system
+    group system
+    interface android.hardware.dumpstate@1.0::IDumpstateDevice default
+    interface android.hardware.dumpstate@1.1::IDumpstateDevice default
+    oneshot
+    disabled
diff --git a/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.xml b/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.xml
new file mode 100644
index 000000000..775e8c841
--- /dev/null
+++ b/dumpstate/1.1/default/android.hardware.dumpstate@1.1-service.xml
@@ -0,0 +1,11 @@
+<manifest version="1.0" type="device">
+    <hal format="hidl">
+        <name>android.hardware.dumpstate</name>
+        <transport>hwbinder</transport>
+        <version>1.1</version>
+        <interface>
+            <name>IDumpstateDevice</name>
+            <instance>default</instance>
+        </interface>
+    </hal>
+</manifest>
-- 
2.21.0

